sql/src/main/java/io/crate/action/sql/query/SortSymbolVisitor.java

```diff
@@ -115,8 +111,38 @@ public SortField visitReference(final Reference symbol, final SortSymbolContext
         // this is why we use a custom comparator source with the same logic as ES
 
         ColumnIdent columnIdent = symbol.info().ident().columnIdent();
-        if (columnIdent.isColumn() && SortParseElement.SCORE_FIELD_NAME.equals(columnIdent.name())) {
-            return !context.reverseFlag ? SortParseElement.SORT_SCORE_REVERSE : SortParseElement.SORT_SCORE;
+
+        if (columnIdent.isColumn()) {
+            if (SortParseElement.SCORE_FIELD_NAME.equals(columnIdent.name())) {
+                return !context.reverseFlag ? SortParseElement.SORT_SCORE_REVERSE : SortParseElement.SORT_SCORE;
+            } else if (DocSysColumns.RAW.equals(columnIdent) || DocSysColumns.ID.equals(columnIdent)) {
+                String luceneName = DocSysColumns.nameForLucene(columnIdent);
+                CollectInputSymbolVisitor.Context inputContext = inputSymbolVisitor.extractImplementations(symbol);
+                ArrayList<Input<?>> inputs = inputContext.topLevelInputs();
+                assert inputs.size() == 1;
+                final Input functionInput = inputs.get(0);
+                @SuppressWarnings("unchecked")
+                final List<LuceneCollectorExpression> expressions = inputContext.docLevelExpressions();
+
+                return new SortField(luceneName, new IndexFieldData.XFieldComparatorSource() {
```

Comment:
would it make sense to create an inner static class for the anonymous XFieldComparatorSource? The one that is created for Function symbols looks pretty much the same